packages/app/src/cli/services/dev/ui.test.tsx

```diff
@@ -154,6 +157,58 @@ describe('ui', () => {
       )
     })
 
+    test("enable dev preview when terminal doesn't support TTY and the app supports it", async () => {
+      vi.mocked(terminalSupportsRawMode).mockReturnValue(false)
+      const concurrentProcess = {
+        prefix: 'prefix',
+        action: vi.fn(async (_stdout, _stderr, _signal) => {}),
+      }
+
+      const processes = [concurrentProcess]
+      const previewUrl = 'https://lala.cloudflare.io/'
+      const graphiqlUrl = 'https://lala.cloudflare.io/graphiql'
+      const app = {
+        canEnablePreviewMode: true,
+        developmentStorePreviewEnabled: false,
+        apiKey: '123',
+        token: '123',
+      }
+
+      const abortController = new AbortController()
+
+      await renderDev({processes, previewUrl, graphiqlUrl, app, abortController})
+      abortController.abort()
+
+      expect(enableDeveloperPreview).toHaveBeenCalledOnce()
+      expect(disableDeveloperPreview).toHaveBeenCalledOnce()
+    })
+
+    test("not enable dev preview when terminal doesn't support TTY and the app does not supports it", async () => {
+      vi.mocked(terminalSupportsRawMode).mockReturnValue(false)
+      const concurrentProcess = {
+        prefix: 'prefix',
+        action: vi.fn(async (_stdout, _stderr, _signal) => {}),
+      }
+
+      const processes = [concurrentProcess]
+      const previewUrl = 'https://lala.cloudflare.io/'
+      const graphiqlUrl = 'https://lala.cloudflare.io/graphiql'
+      const app = {
+        canEnablePreviewMode: false,
+        developmentStorePreviewEnabled: false,
+        apiKey: '123',
+        token: '123',
+      }
+
+      const abortController = new AbortController()
+
+      await renderDev({processes, previewUrl, graphiqlUrl, app, abortController})
+      abortController.abort()
+
+      expect(enableDeveloperPreview).not.toHaveBeenCalledOnce()
+      expect(disableDeveloperPreview).not.toHaveBeenCalledOnce()
```

Comment:
I know it's nitpicky, but wouldn't this be better as:

```suggestion
      expect(enableDeveloperPreview).not.toHaveBeenCalled()
      expect(disableDeveloperPreview).not.toHaveBeenCalled()
```

I could be wrong, but maybe `not.toHaveBeenCalledOnce()` would be truthy if it were called twice. Removing the `Once` prevents this ambiguity.